File: programming-concepts/selection/python/show_error_case.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def show_error():
    """Example of a match/case statement to match the error code"""
    error_code = input("Enter the error code: ")
    
    match error_code:
        case "400":
            print("Bad request")
        case "401":
            print("Unauthorised request")
        case "403":
            print("Forbidden request")
        case "404":
            print("Page not found")
        case "408":
            print("Timeout error")
        case _:
            print("Unknown error")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    show_error()

File: programming-concepts/selection/c-sharp/show_error_case.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class Selection
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            ShowError();
        }


        // Example of a switch/case statement to match the error code
        public static void ShowError() {
            Console.WriteLine("Enter the error code: ");
            string errorCode = Console.ReadLine();

            switch (errorCode) {
                case "400":
                    Console.WriteLine("Bad request");
                    break;
                case "401":
                    Console.WriteLine("Unauthorised request");
                    break;
                case "403":
                    Console.WriteLine("Forbidden request");
                    break;
                case "404":
                    Console.WriteLine("Page not found");
                    break;
                case "408":
                    Console.WriteLine("Timeout error");
                    break;
                default:
                    Console.WriteLine("Unknown error");
                    break;
            }
        }
        

    }
}
